packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,169 @@
+    --background-hover-color: var(--onyx-color-base-primary-400, #3b98d9);
+    --background-disabled-color: var(--onyx-color-base-primary-200, #bfddf3);
+    --color: var(--onyx-color-text-icons-neutral-inverted, #fff);
+    --disabled-color: var(--onyx-color-text-icons-neutral-inverted, #fff);
+    --border-color: var(--onyx-color-base-primary-500, #0079ce);
+    --disabled-border-color: var(--onyx-color-base-primary-200, #bfddf3);
+  }
+
+  &--default--secondary {
+    --background-color: var(--onyx-color-base-background-blank, #fff);
+    --background-hover-color: var(--onyx-color-base-neutral-200, #eaedf0);
+    --background-disabled-color: var(--onyx-color-base-background-blank, #fff);
+    --border-color: var(--onyx-color-base-neutral-400, #b7bfc5);
+    --disabled-border-color: var(--onyx-color-base-neutral-200, #eaedf0);
+  }
+
+  &--default--danger {
+    --background-color: var(--onyx-color-base-danger-200, #f1d3d2);
+    --background-hover-color: var(--onyx-color-base-danger-100, #fbefee);
+    --background-disabled-color: var(--onyx-color-base-danger-100, #fbefee);
+    --border-color: var(--onyx-color-base-danger-500, #d1332f);
+    --disabled-border-color: var(--onyx-color-base-danger-200, #f1d3d2);
+  }
+
+  &--outline--primary {
+    --border-color: var(--onyx-color-base-primary-500, #0079ce);
+    --disabled-border-color: var(--onyx-color-base-primary-200, #bfddf3);
+  }
+
+  &--outline--secondary {
+    --border-color: var(--onyx-color-base-neutral-400, #b7bfc5);
+    --disabled-border-color: var(--onyx-color-base-neutral-200, #eaedf0);
+  }
+
+  &--outline--danger {
+    --border-color: var(--onyx-color-base-danger-500, #d1332f);
+    --disabled-border-color: var(--onyx-color-base-danger-200, #f1d3d2);
+  }
+
+  background-color: var(--background-color);
+  border: 0.063rem solid var(--border-color);
+
+  &:hover {
+    background: var(--background-hover-color);
+  }
+
+  &:focus-visible {
+    outline: 0.25rem solid var(--focus-outline-color);
+  }
+
+  &:has(&__element:disabled) {
+    cursor: none;
+    pointer-events: none;
+    background-color: var(--background-disabled-color);
+    border-color: var(--disabled-border-color);
+
+    .onyx-button__element__label {
+      color: var(--disabled-color);
+    }
+  }
+
+  &__element {
+    position: relative;
+    margin: 0;
+    padding: 0;
+    border: none;
+    cursor: pointer;
+    background: transparent;
+
+    &:focus-visible {
+      outline: none;
+    }
+
+    &__label {
+      display: flex;
+      padding: 0rem var(--onyx-spacing-4xs, 0.25rem);
+      justify-content: center;
+      align-items: center;
+      gap: 0.625rem;
+      max-width: 12.25rem;
+      color: var(--color);
+      -webkit-box-orient: vertical;
+      -webkit-line-clamp: 1;
+      overflow: hidden;
+      text-overflow: ellipsis;
+      font-size: 1rem;
+      font-style: normal;
+      font-weight: 600;
+      line-height: 1.5rem;
+    }
```

Comment:
```suggestion
    &__label {
      display: inline-block;
      padding: 0 var(--onyx-spacing-4xs);
      max-width: 12.25rem;
      color: var(--color);
      overflow: hidden;
      text-overflow: ellipsis;
      font-size: 1rem;
      font-weight: 600;
      line-height: 1.5rem;
    }
```

Not all Figma properties are relevant, we should only keep the ones that are actually needed